homeassistant/components/enocean/sensor.py

```diff
@@ -40,37 +44,49 @@
 SENSOR_TYPE_TEMPERATURE = "temperature"
 SENSOR_TYPE_WINDOWHANDLE = "windowhandle"
 
-SENSOR_DESC_TEMPERATURE = SensorEntityDescription(
+
+@dataclass
+class EnOceanSensorEntityDescription(SensorEntityDescription):
+    """Sensor description for EnOcean sensors with additional unique id."""
+
+    unique_id: Callable[[list[int]], str | None] = None  # type: ignore[assignment]
```

Comment:
Please create a mixin dataclass so we can make this attribute required. See the example I linked.

Reply:
Two questions:
1) Is the purpose of the mixin class to create an abstract placeholder for any future attributes to come?
2) I don't have a use case for exists_fn... ? Wouldn't I implement dead code here?

Thanks for your help!

Reply:
The purpose of the mixin dataclass is to be able to add an attribute to the entity description that is required. We can't add required attributes in subclasses of dataclass after optional attributes in the parent. The mixin dataclass solves that limitation.

I'm not asking you to add the attribute `exists_fn`. The point of the example is to show how to add a mixin dataclass with a required attribute and use the mixin dataclass in the entity description subclass.

Reply:
Done. I hope that works now ... 🤞🏼